metaflow/plugins/kfp/tests/.gitlab-ci.yml

```diff
@@ -1,211 +1,142 @@
-
-    # Function definitions for logging and uploading to docker repository
-    - 'function log_upload_path {
-        DOCKER_IMAGE_NAME=$1;
-        DOCKER_IMAGE_TAG=$2;
-        echo "Project Image Repository -" $(echo ${ARTIFACTORY_WEB_APP_URL}/$(echo "${DOCKER_IMAGE_NAME}" | sed "s/\//~2F/g" | xargs -0 printf "%b"));
-      }'
-
-    - 'function docker_push {
-           LOCAL_DOCKER_TAG=$1;
-           DOCKER_IMAGE_NAME=$2;
-           DOCKER_IMAGE_TAG=$3;
-           export DOCKER_REPO_UPLOAD_PATH=${DOCKER_REPO_URL}/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG};
-
-           docker tag ${LOCAL_DOCKER_TAG} ${DOCKER_REPO_UPLOAD_PATH};
-           docker push ${DOCKER_REPO_UPLOAD_PATH};
-           log_upload_path ${DOCKER_IMAGE_NAME} ${DOCKER_IMAGE_TAG};
-       }'
-
-    # Login to docker
-    - 'echo ${DOCKER_API_KEY} | docker login -u ${DOCKER_USERNAME} --password-stdin ${DOCKER_REPO_URL}'
+stages:
+  - build
+  - test
 
-    # Push to artifactory repo (overwrites are NOT allowed to docker-images in this repo)
-    - 'echo "Will run docker push" ${_LOCAL_DOCKER_TAG} ${_EXPANDED_VERSION_DOCKER_IMAGE_NAME} ${_EXPANDED_VERSION_DOCKER_IMAGE_TAG}'
-    - STDERR=$(docker_push ${_LOCAL_DOCKER_TAG} ${_EXPANDED_VERSION_DOCKER_IMAGE_NAME} ${_EXPANDED_VERSION_DOCKER_IMAGE_TAG} 2>&1 >/dev/null) || true
-    - echo $STDERR
-    - |
-      if [ -z "$STDERR" ] || [[ $STDERR == *"manifest invalid"* ]]; then
-        echo "Successful!"
-      else
-        echo "Error in pushing image to Artifactory."
-        exit 1
-      fi
-  
-    - 'echo "built-image-url: ${DOCKER_REPO_URL}/${_EXPANDED_VERSION_DOCKER_IMAGE_NAME}:${_EXPANDED_VERSION_DOCKER_IMAGE_TAG}"'
-   
 .path_to_host_file_directory: # obtains the path to file on the host machine, which is needed to mount a Docker on Docker
   script: &path-to-host-file-directory
   - export CONTAINER_ID=$(docker ps -q -f "label=com.gitlab.gitlab-runner.job.id=$CI_JOB_ID" -f "label=com.gitlab.gitlab-runner.type=build")
   - export MOUNT_NAME=$(docker inspect $CONTAINER_ID -f "{{ range .Mounts }}{{ if eq .Destination \"/builds\" }}{{ .Source }}{{end}}{{end}}")
   - export SHARED_PATH=$MOUNT_NAME/$CI_PROJECT_PATH
-  
-build:
-  stage: build
-  script:
-    - *init-docker-variables
-    - *build-and-push-docker-image
-    - export BUILT_IMAGE_FULL_PATH=${DOCKER_REPO_URL}/${_EXPANDED_VERSION_DOCKER_IMAGE_NAME}:${_EXPANDED_VERSION_DOCKER_IMAGE_TAG}
-    - echo ${BUILT_IMAGE_FULL_PATH} > ${_IMAGE_TAG_FILE} # file _IMAGE_TAG_FILE contains the path to the image
-  artifacts:
-    paths:
-      - ${_IMAGE_TAG_FILE} # we pass this file as an artifact to be read in the testing stages
 
-test:internal:
-  extends: 
-    - .devex_internal_eks # sets up AWS environment variables
-    - .generate_kubeconfig # creates the .kube directory for the appropriate cluster
+.version: &version |
+  METAFLOW_VERSION=$(cat $METAFLOW_VERSION_PATH | sed -nr "s/^version = ['\"]([^'\"]*)['\"]$/\1/p")
```

Comment:
Do you mind adding a quick comment on what this regular expression does? It's pretty unruly!

Reply:
Yeah sure added, effectively it's just grabbing the version string, so from `version = 'x.y.z'` `METAFLOW_VERSION` would equal `x.y.z` after this regex 😄